destination/src/main/scala/quasar/plugin/sqlserver/destination/SinkBuilder.scala

```diff
@@ -189,4 +189,20 @@ object SinkBuilder {
       } yield Stream.eval_(logStart) ++ Stream.eval_(start) ++ handled ++ Stream.eval_(logEnd)
     }
   }
+
+  /** Ensures the provided identity column is suitable for indexing by SQL Server,
+    * adjusting the type to one that is compatible and indexable if not.
+    */
+  private def ensureIndexableIdColumn(
+      id: Column[SQLServerType],
+      columns: NonEmptyList[Column[SQLServerType]])
+      : (Column[SQLServerType], NonEmptyList[Column[SQLServerType]]) =
+    Typer.inferScalar(id.tpe)
+      .collect {
+        case t @ ColumnType.String if Some(id.tpe) == Typer.preferred(t) =>
```

Comment:
Why is this `==`, not `===`?

What would happen if user sets `NVARCHAR` with length greater than max indexable but lesser than maximum available? E.g. `VARCHAR(2048)` 

Maybe use `NVARCHAR`?

Reply:
Thanks re `==`, that was an oversight will fix (I've been writing a lot of `==` elsewhere recently 😃).

As for the behavior with `NVARCHAR` or other lengths, the high-level goal is to only perform this automatic adjustment if the column is coerced to the "default" `SQLServerType` for the `ColumnType`. A more robust way of solving this, would be to make the "default" coercion a first-class concept that is communicated to the destination (i.e. one could choose an explicit type or "default" which allows the destination choose something "appropriate"), but that has the potential to affect a lot of destinations and this was a time-sensitive fix.

If a user explicitly specified a type for the column, we don't want to mess with it, if it errors then they can fix it since they changed it in the first place. But if they just go with the defaults, we're ok adjusting to make sure things work.

Reply:
Got it, since we only have `TEXT` by default, we need to convert it to `VARCHAR`, if it would be `NTEXT` the target was `NVARCHAR` :+1: